tensorflow/core/kernels/reduction_gpu_kernels.cu.h

```diff
@@ -388,8 +388,19 @@ __global__ __launch_bounds__(1024) void ColumnReduceKernel(
     //  #         #  block boundary
     //  -         =
     //            =
+#ifdef _MSC_VER
+#if _MSC_VER >= 1930
+    const int numRowsThisBlock =
+        min(blockDim.y, num_rows - blockIdx.y * blockDim.y);
```

Comment:
Why would some versions of msvc require a cast, and some not?

What is the type of `num_rows - blockIdx.y * blockDim.y`?  This shouldn't change based on the compiler.  The first argument should be the same type so we are comparing like kinds.